pkg/key/key.go

```diff
@@ -0,0 +1,9 @@
+package key
+
+const (
+	CalicoVersionConstraint                       = ">= 3.10.0 < 3.15.0"
+	EtcdVersionConstraint                         = ">= 3.4.0 < 3.5.0"
```

Comment:
Are these constraints universally valid? I could see that we have quite different requirements across providers. Also IIRC for HA Masters we needed `3.4.10` or something like that. Maybe Mr @calvix can check back on this. I would further appreciate some godoc to explain what these constants are about just to provide more context.

Reply:
These constraints should allow etcd `3.4.10`. I did want to open this up to all of SIG Operator to make sure I wasn't missing any use cases though--just let me know. And I'll add some godoc comments in a bit.

Reply:
I added a comment for this block of constants.

Reply:
I checked AWS 11.5.0 and 12.0.0 and this constraint should work for both (etcd 3.4.9 and 3.4.10 respectively).